Formula/gcc.rb

```diff
@@ -232,6 +232,16 @@ def post_install
 
       EOS
       inreplace(specs, " %o ", "\\0%(homebrew_rpath) ")
+    elsif OS.mac?
+      # libstdc++: Fix iostream init for Clang on Darwin
+      # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=110432#c7
+      # https://gcc.gnu.org/git/gitweb.cgi?p=gcc.git;h=fe2651affa8c15624188bfd062fb894648743431
+      if version >= "13.2.0"
+        odie "Review if the fix is still required."
+      else
+        inreplace "/usr/local/opt/gcc@13/include/c++/13/iostream",
+          /^(#if !__has_attribute\(__init_priority__\))$/, '\1 || defined __APPLE__'
+      end
```

Comment:
Let's not do this in `post_install`, please. Particularly since we are calling `odie`, this should happen as early in the install process as possible.

Reply:
Probably this could also be done differently without `odie`. I just wanted to assure that this fix gets removed for the GCC 13.2 release, since this should be fixed then.

Not sure why `post_install` is not good since this will have no side effect for GCC itself.

Reply:
Would the bottles needed to be recreated if we put it in `install`? This is definitely not needed for the fix.

Reply:
@carlocab To avoid the `odie` I could also change the if condition to `if version >= "13.1.0" && version < "13.2.0"

```
if version >= "13.1.0" && version < "13.2.0
  inreplace "/usr/local/opt/gcc@13/include/c++/13/iostream",
    /^(#if !__has_attribute\(__init_priority__\))$/, '\1 || defined __APPLE__'
end
```

Would this be better then? I thought to add the `odie` that this part gets detected more easily and then can be removed. The `odie` should  never happen to a normal user. Only (probably) when a maintainer updates to the GCC >= 13.2.